rigging/__init__.py

```diff
@@ -482,10 +393,9 @@ def _destroy_self(self):
         """
         Called when this rig receives a destroy command from the CLI
 
-        :return: DestroyRig exception
         """
         self.logger.info(
             'Received destroy command, terminating rig without '
             'triggering actions'
         )
-        raise DestroyRig
+        self._exit(0)
```

Comment:
Hmm. This is both functional and simple, but I'm wondering if there are any edge cases where we'd want to terminate just the thread (rather than the whole process) and let that propagate through the normal handling methods?

Reply:
There are. I ended up calling `self._exit(0)` here just trying to mimic what raising `DestroyRig` produced.  

I had to change the `RigDBusListener.destroy()` definition/behavior so the response is provided via dbus async callbacks instead of just the return value, otherwise when running the `destroy()` callback the process would exit before the method returned, hence no feedback message was sent back to the client.  
Ending just the thread running the glib bus would have the same side effects though.  

You'll probably be more familiar with proper ways to cleanup everything and exit when destroy is called. If you have other idea about how should destroy behave so the shutdown procedure is cleaner that would be great.